br/pkg/lightning/restore/restore.go

```diff
@@ -1197,37 +1199,64 @@ func (rc *Controller) buildRunPeriodicActionAndCancelFunc(ctx context.Context, s
 						state = "preparing"
 					}
 
-					// since we can't accurately estimate the extra time cost by import after all writing are finished,
-					// so here we use estimatedWritingProgress * 0.8 + estimatedImportingProgress * 0.2 as the total
-					// progress.
+					// lightning restore is separated into restore engine and import engine, they are both parallelized
+					// and pipelined between engines, so we can only weight the progress of those 2 phase to get the
+					// total progress.
+					//
+					// for local & importer backend:
+					// in most case import engine is faster since there's little computations, but inside one engine
+					// restore and import is serialized, the progress of those two will not differ too much, and
+					// import engine determines the end time of the whole restore, so we average them for now.
```

Comment:
Although the import is the last stage, why is it an average and not 8:2 or 7:3 or 6:4? Import is usually faster that writing kvs to local storage.

Reply:
for inside one engine, restore(encode&deliver) and import(write and ingest) is serialized, the progress of those two will not differ too much and import is faster in most case.